code/_onclick/observer.dm

```diff
@@ -14,7 +14,7 @@
 
 	// Otherwise jump
 	else if(A.loc)
-		loc = get_turf(A)
+		forceMove(get_turf(A))

```

Comment:
I would like to point out a thing #33461
This will also happen for ghosts, and you should probably fix it.